integration-cli/docker_cli_login_test.go

```diff
@@ -10,20 +9,22 @@ import (
 func TestLoginWithoutTTY(t *testing.T) {
 	cmd := exec.Command(dockerBinary, "login")
 
-	// create a buffer with text then a new line as a return
-	buf := bytes.NewBuffer([]byte("buffer test string \n"))
-
 	// use a pipe for stdin and manually copy the data so that
 	// the process does not get the TTY
 	in, err := cmd.StdinPipe()
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	if err := cmd.Start(); err != nil {
+		t.Fatal(err)
+	}
+
 	// copy the bytes into the commands stdin along with a new line
-	go io.Copy(in, buf)
+	io.WriteString(in, "buffer test string \n")
```

Comment:
I wonder if this can be just `cmd.Stdin = bytes.NewBufferString("buffer test string \n")` and `Run`